zebra-state/src/service/check.rs

```diff
@@ -416,3 +491,225 @@ pub(crate) fn initial_contextual_validity(
+            .constrain::<NegativeAllowed>()
+            .map_err(|e| SubsidyError::Other(format!("invalid deferred pool amount: {e}")))?;
+
+        // Checks the one-time lockbox disbursements in the NU6.1 activation block's coinbase transaction
+        // See [ZIP-271](https://zips.z.cash/zip-0271) and [ZIP-1016](https://zips.z.cash/zip-1016) for more details.
+        let expected_one_time_lockbox_disbursements = network.lockbox_disbursements(height);
+        for (address, expected_amount) in &expected_one_time_lockbox_disbursements {
+            if !has_expected_output(address, *expected_amount) {
+                Err(SubsidyError::OneTimeLockboxDisbursementNotFound)?;
+            }
+
+            deferred_pool_balance_change = deferred_pool_balance_change
+                .checked_sub(*expected_amount)
+                .expect("should be a valid Amount");
+        }
+
+        // # Consensus
+        //
+        // > [Canopy onward] The coinbase transaction at block height `height`
+        // > MUST contain at least one output per funding stream `fs` active at `height`,
+        // > that pays `fs.Value(height)` zatoshi in the prescribed way to the stream's
+        // > recipient address represented by `fs.AddressList[fs.AddressIndex(height)]
+        //
+        // https://zips.z.cash/protocol/protocol.pdf#fundingstreams
+        for (receiver, expected_amount) in funding_streams {
+            let address = funding_stream_address(height, network, receiver)
+                // funding stream receivers other than the deferred pool must have an address
+                .ok_or_else(|| {
+                    SubsidyError::Other(format!(
+                        "missing funding stream address at height {height:?}"
+                    ))
+                })?;
+
+            if !has_expected_output(address, expected_amount) {
+                Err(SubsidyError::FundingStreamNotFound)?;
+            }
+        }
+
+        Ok(DeferredPoolBalanceChange::new(deferred_pool_balance_change))
+    } else {
+        // Future halving, with no founders reward or funding streams
+        Ok(DeferredPoolBalanceChange::zero())
+    }
+}
+
+/// Returns `Ok(())` if the miner fees consensus rule is valid.
+///
+/// [7.1.2]: https://zips.z.cash/protocol/protocol.pdf#txnconsensus
+pub fn miner_fees_are_valid(
+    coinbase_tx: &Transaction,
+    height: Height,
+    block_miner_fees: Amount<NonNegative>,
+    expected_block_subsidy: Amount<NonNegative>,
+    expected_deferred_pool_balance_change: DeferredPoolBalanceChange,
+    network: &Network,
+) -> Result<(), SubsidyError> {
+    let transparent_value_balance = coinbase_tx
+        .outputs()
+        .iter()
+        .map(|output| output.value())
+        .sum::<Result<Amount<NonNegative>, AmountError>>()
+        .map_err(|_| SubsidyError::SumOverflow)?
+        .constrain()
+        .map_err(|e| SubsidyError::Other(format!("invalid transparent value balance: {e}")))?;
```

Comment:
```suggestion
        .map(|sum| sum.constrain())
        .map_err(|_| SubsidyError::SumOverflow)?;
```
or if that doesn't work,
```suggestion
        .map_err(|_| SubsidyError::SumOverflow)?
        .constrain()
        .map_err(|_| SubsidyError::SumOverflow)?;
```